[WIP] OBSDOCS-1079: Create a table listing and describing all RBAC permissi… (#80127)
openshift/openshift-docs: modules/monitoring-granting-users-permissions-for-core-platform-monitoring.adoc

// Module included in the following assemblies:
//
// * observability/monitoring/configuring-the-monitoring-stack.adoc

:_mod-docs-content-type: CONCEPT
[id="granting-users-permissions-for-core-platform-monitoring_{context}"]
= Granting users permissions for core platform monitoring

As a cluster administrator, you can monitor all core {product-title} and user-defined projects.

You can also grant developers and other users different permissions for core platform monitoring:

* To monitor core platform monitoring components.
* To query the cluster resource metrics API.

> * To monitor or manage alerts and silences.

Reviewer: (nit) "and" rather than "or"? or "To manage alerts and silences"?

> * To access Alertmanager component.

Reviewer: Alertmanager = silences?

> You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

Reviewer: should we state explicitly if a row is a role or cluster role?

Author: I hope it will be clear with the addition of the third column here :+1:

|===
|Name |Description

> |`cluster-monitoring-metrics-api` |Users with this role have the ability to query the cluster resource metrics API for core platform monitoring.

Reviewer: should we refer to https://docs.openshift.com/container-platform/4.17/observability/monitoring/accessing-third-party-monitoring-apis.html#openshift-monitoringthanos-querier-2 and https://docs.openshift.com/container-platform/4.17/observability/monitoring/accessing-third-party-monitoring-apis.html#openshift-monitoringthanos-querier?
After checking the code, it also allows access to user-defined Prometheus and Thanos Ruler API endpoints.
```suggestion
|`cluster-monitoring-metrics-api` |Users with this role have the ability to query Thanos Querier web server on port 9091.
```

Author: As the current docs only mention `cluster-monitoring-view` as a required cluster role to be able to access this port, should I also add to [these docs](https://docs.openshift.com/container-platform/4.17/observability/monitoring/accessing-third-party-monitoring-apis.html#openshift-monitoringthanos-querier-2) that the access is also available with the `cluster-monitoring-metrics-api` role?

Reviewer: Yes we'd need to update these docs too. The source is [there](https://github.com/openshift/cluster-monitoring-operator/blob/master/Documentation/resources.adoc) so I can fix it here first.

Author: Oh great, thanks! I just noticed that we generate that part of docs automatically, so it would be great if you fixed it there :+1:

> |`cluster-monitoring-operator-alert-customization` |Users with this role can monitor `AlertingRule` and `AlertRelabelConfig` resources for core platform monitoring. These permissions are required for the alert customization feature.

Reviewer: for me, "monitor" implies read-only access while the role gives all permissions on the mentioned resources.
```suggestion
|`cluster-monitoring-operator-alert-customization` |Users with this role can manage `AlertingRule` and `AlertRelabelConfig` resources for core platform monitoring. These permissions are required for the alert customization feature.
```

> |`monitoring-alertmanager-edit` |Users with this role can create and silence alerts for core platform monitoring in the *Administrator* perspective of the {product-title} web console.

Reviewer: It also provides access to the Alertmanager web server on port 9094 https://docs.openshift.com/container-platform/4.17/observability/monitoring/accessing-third-party-monitoring-apis.html#openshift-monitoringalertmanager-main
```suggestion
|`monitoring-alertmanager-edit` |Users with this role can manage alert silences for core platform monitoring in the *Administrator* perspective of the {product-title} web console.
```

> |`monitoring-alertmanager-view` |Users with this role have read access to the Alertmanager API for core platform monitoring. They can also view alert silences in the *Developer* perspective of the {product-title} web console.

Reviewer: IIRC access to silences in the developer console is governed by `monitoring-rules-edit` or `monitoring-edit`.
```suggestion
|`monitoring-alertmanager-view` |Users with this role have read access to the Alertmanager API for core platform monitoring. They can also view alert silences in the *Administrator* perspective of the {product-title} web console.
```

> |`cluster-monitoring-view` |Users with this cluster role can access Alertmanager component for core platform monitoring.

Reviewer: this one is a bit complicated:
* It's a cluster role which requires a ClusterRoleBinding instead of RoleBinding like the other roles above.
* In practice it's equivalent to `cluster-monitoring-metrics-api` with additional permissions on the `/federate` endpoint of the user-defined Prometheus.

// Module included in the following assemblies:
//
// * observability/monitoring/configuring-the-monitoring-stack.adoc

:_mod-docs-content-type: CONCEPT
[id="granting-users-permissions-for-core-platform-monitoring_{context}"]
= Granting users permissions for core platform monitoring

As a cluster administrator, you can monitor all core {product-title} and user-defined projects.

> You can also grant developers and other users different permissions for core platform monitoring. You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

Reviewer 2: Should this be _or_, as the user selects only one?
 ```suggestion
You can also grant developers and other users different permissions for core platform monitoring. You can grant the permissions by assigning one of the following monitoring roles or cluster roles:
```
or, do you grant one monitoring role **and** one cluster role?

|`openshift-monitoring`

|`cluster-monitoring-operator-alert-customization` 
|Users with this role can manage `AlertingRule` and `AlertRelabelConfig` resources for core platform monitoring. These permissions are required for the alert customization feature. 
|`openshift-monitoring`

|`monitoring-alertmanager-edit` 
|Users with this role can manage the Alertmanager API for core platform monitoring. They can also manage alert silences in the *Administrator* perspective of the {product-title} web console.
|`openshift-monitoring`

|`monitoring-alertmanager-view` 
|Users with this role can monitor the Alertmanager API for core platform monitoring. They can also view alert silences in the *Administrator* perspective of the {product-title} web console. 
|`openshift-monitoring`

|`cluster-monitoring-view` 
|Users with this cluster role have the same access rights as `cluster-monitoring-metrics-api` role, with additional permissions, providing access to the `/federate` endpoint for the user-defined Prometheus.

> |Has to be bound with `ClusterRoleBinding` to gain access to the `/federate` endpoint for the user-defined Prometheus.

Reviewer 2: ```suggestion
|Must be bound with `ClusterRoleBinding` to gain access to the `/federate` endpoint for the user-defined Prometheus.
```

// Module included in the following assemblies:
//
// * observability/monitoring/configuring-the-monitoring-stack.adoc

:_mod-docs-content-type: CONCEPT
[id="granting-users-permissions-for-core-platform-monitoring_{context}"]
= Granting users permissions for core platform monitoring

As a cluster administrator, you can monitor all core {product-title} and user-defined projects.

You can also grant developers and other users different permissions for core platform monitoring:

* To monitor core platform monitoring components.
* To access the Thanos Querier API endpoints.
* To manage alert silences.

> * To monitor the Alertmanager API.

Reviewer: (discussion) I feel that we mix and match high-level functionality ("monitor core platform monitoring components") with low-level stuff ("access Thanos Querier API"). Should we avoid implementation details and leave that for the role descriptions?

Author: Yes, that makes sense. I tried to create something similar to what is in the UWM section, but here it is a bit awkward. I aggree that it would be probably better to remove it :+1:  Thank you!

> You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

Reviewer: (discussion) Now that we have a split between role and clusterrole in the other table, should we do the same here?

Author: I did not want to do this here, because we only list one cluster role and implementing a table with just one role is not ideal :sweat_smile: . So I think having it in one table (at least for now) is a better choice. :+1:

|===
|Name |Description |Project

|`cluster-monitoring-metrics-api` 

> |Users with this role have the ability to access Thanos Querier API endpoints. Additionally, it grants access to user-defined Prometheus and Thanos Ruler API endpoints. 

Reviewer: there's no way to access the user-defined Prometheus API.
It also provides access to the Administrator metrics and alerts pages.
```suggestion
|Users with this role have the ability to access Thanos Querier API endpoints. Additionally, it grants access to the core platform Prometheus API and user-defined Thanos Ruler API endpoints. 
```

|`openshift-monitoring`

|`cluster-monitoring-operator-alert-customization` 
|Users with this role can manage `AlertingRule` and `AlertRelabelConfig` resources for core platform monitoring. These permissions are required for the alert customization feature. 
|`openshift-monitoring`

|`monitoring-alertmanager-edit` 
|Users with this role can manage alert silences for core platform monitoring in the *Administrator* perspective of the {product-title} web console. Additionally, it provides access to the Alertmanager web server on port 9094. 
|`openshift-monitoring`

|`monitoring-alertmanager-view` 

> |Users with this role can monitor the Alertmanager API for core platform monitoring. They can also view alert silences in the *Administrator* perspective of the {product-title} web console. 

Reviewer: the wording isn't consistent with 'monitoring-alertmanager-edit'. Should we fix?

|`openshift-monitoring`

|`cluster-monitoring-view` 
|Users with this cluster role have the same access rights as `cluster-monitoring-metrics-api` role, with additional permissions, providing access to the `/federate` endpoint of the user-defined Prometheus. 

> |Can be bound with `RoleBinding` to any user project.

Reviewer: It should be bound via a ClusterRoleBinding.
